src/gui/waveEdit.cpp

```diff
@@ -480,84 +726,239 @@ void FurnaceGUI::drawWaveEdit() {
                   if (CWSliderInt("##WGFB",&waveGenFB[i],0,7)) {
                     doGenerateWave();
                   }
+                  
                   ImGui::PopID();
                 }
 
                 ImGui::EndTable();
               }
 
+
+
+              if (ImGui::BeginTable("WGFMWAVE", 2)) {
+                  ImGui::TableSetupColumn("c0", ImGuiTableColumnFlags_WidthFixed, ImGui::CalcTextSize("Op").x);
+                  ImGui::TableSetupColumn("c1", ImGuiTableColumnFlags_WidthStretch, 1);
+
+                  ImGui::TableNextRow(ImGuiTableRowFlags_Headers);
+                  ImGui::TableNextColumn();
+                  ImGui::Text("Op");
+                  ImGui::TableNextColumn();
+                  ImGui::Text("Waveform");
+
+                  for (int i = 0; i < 4; i++) {
+                      ImGui::TableNextRow();
+                      ImGui::TableNextColumn();
+                      ImGui::Text("%d", i + 1);
+
+                      ImGui::TableNextColumn();
+                      ImGui::SetNextItemWidth(ImGui::GetContentRegionAvail().x);
+                      ImGui::PushID(i);
+                      if (CWSliderInt("##WGWAVEFORM", &fmWaveform[i], 0, _countof(fmWaveforms)-1, fmWaveforms[fmWaveform[i]])) {
```

Comment:
Don't use _countof. Instead define fmWaveformsLen after the definition of fmWaveforms.....

I'll fix this myself; don't worry....